Testscripts/Linux/SetupHbKernel.sh

```diff
@@ -141,18 +152,114 @@ function Main() {
 			LogMsg "$?: Added resume=$sw_uuid in /etc/default/grub file"
 		fi
 
+		_entry=$(cat /etc/default/grub | grep 'GRUB_HIDDEN_TIMEOUT=')
+		if [ -n "$_entry" ]; then
+			sed -i -e "s/GRUB_HIDDEN_TIMEOUT=*.*/GRUB_HIDDEN_TIMEOUT=30/g" /etc/default/grub
+			LogMsg "$?: Updated GRUB_HIDDEN_TIMEOUT value with 30"
+		else
+			echo 'GRUB_HIDDEN_TIMEOUT=30' >> /etc/default/grub
+			LogMsg "$?: Added GRUB_HIDDEN_TIMEOUT=30 in /etc/default/grub file"
+		fi
+
+		_entry=$(cat /etc/default/grub | grep 'GRUB_TIMEOUT=')
+		if [ -n "$_entry" ]; then
+			sed -i -e "s/GRUB_TIMEOUT=.*/GRUB_TIMEOUT=30/g" /etc/default/grub
+			LogMsg "$?: Updated GRUB_TIMEOUT value with 30"
+		else
+			echo 'GRUB_TIMEOUT=30' >> /etc/default/grub
+			LogMsg "$?: Added GRUB_TIMEOUT=30 in /etc/default/grub file"
+		fi
+
+		# VM Gen is 2
 		if [[ "$os_GENERATION" == "2" ]];then
 			grub_cfg="/boot/efi/EFI/redhat/grub.cfg"
 		else
-			grub_cfg="/boot/grub2/grub.cfg"
+			# VM Gen is 1
+			if [ -f /boot/grub2/grub.cfg ]; then
+				grub_cfg="/boot/grub2/grub.cfg"
+			else
+				grub_cfg="/boot/grub/grub.cfg"
+			fi
+		fi
+		# grub2-mkconfig shows the image build problem in RHEL/CentOS, so we use alternative.
+		#grub2-mkconfig -o ${grub_cfg}
+		#LogMsg "$?: Run grub2-mkconfig -o ${grub_cfg}"
+		ls /boot/vmlinuz* > new_state.txt
```

Comment:
Is `/boot` a separate partition? Can we ensure it's mounted before continuing with tests like this?

Reply:
In Azure VM, yes. /boot is a different partition. I think this is general rule of Azure image layout. Without it, VM provision failed before getting into this point. I am not sure we need to check again